src/components/user/AddFile.tsx

```diff
@@ -0,0 +1,226 @@
+                setSelectedTags([...selectedTags, tag]);
+              } else {
+                setSelectedTags(selectedTags.filter((i) => i != tag));
+              }
+            }}
+          />
+          {!selectedTags.includes(tag) && (
+            <label
+              htmlFor={tag.name}
+              className="m-2 ml-[-2px] inline-block rounded-full border-2 p-2 hover:bg-white hover:text-[#22555A]"
+            >
+              {" "}
+              {tag.name}
+            </label>
+          )}
+          {selectedTags.includes(tag) && (
+            <label
+              htmlFor={tag.name}
+              className="m-2 ml-[-2px] inline-block rounded-full border-2 bg-white p-2 text-[#22555A]"
+            >
+              {" "}
+              {tag.name}
+            </label>
+          )}
+        </div>
+      ))}
+    </div>
+  );
+};
+
+const TagSelector = ({
+  selectedTags,
+  setSelectedTags,
+}: {
+  selectedTags: TagProps[];
+  setSelectedTags: React.Dispatch<React.SetStateAction<TagProps[]>>;
+}) => {
+  return (
+    <div>
+      <div className="text-neutral-600 mb-1 h-[34px] w-full font-['merriweather'] text-2xl">
+        Tags
+      </div>
+      <div className="text-lg font-thin">
+        Select min of 1, max of 3 <span className="text-red-400">*</span>
+      </div>
+      <TagOptions
+        selectedTags={selectedTags}
+        setSelectedTags={setSelectedTags}
+      />
+    </div>
+  );
+};
+
+const AddFile = ({
+  showAddFilePopUp,
+  setShowAddFilePopUp,
+  seniorId,
+  files,
+  folder,
+}: AddFileProps) => {
+  const currFiles = Object.values(files);
+  const excludeDates = currFiles.map((fileObj) => fileObj.date);
+  const excludedDatesString = excludeDates.map((dateObj) =>
+    dateObj.toDateString()
+  );
+
+  const startDate_ = new Date();
+
+  while (excludedDatesString.includes(startDate_.toDateString())) {
+    startDate_.setDate(startDate_.getDate() + 1);
+  }
+
+  const [startDate, setStartDate] = useState(startDate_);
+
+  const [confirm, setConfirm] = useState<boolean>(false);
+  const [error, setError] = useState<boolean>(false);
+  const [selectedTags, setSelectedTags] = useState<TagProps[]>([]);
+  const [datePickerOpen, setDatePickerOpen] = useState(false);
+
+  const handleCancel = () => {
+    setShowAddFilePopUp(!showAddFilePopUp);
+  };
+
+  const callAddFile = async () => {
+    const response = await createFile({
+      body: {
+        date: startDate,
+        filetype: "Google Document",
+        url: "",
+        Tags: selectedTags.map((tagProp) => tagProp.name),
+        seniorId: seniorId,
+      },
+    });
+
+    if (response.code == "SUCCESS") {
+      setConfirm(true);
+    } else {
+      setError(true);
+    }
+  };
+
+  return (
+    <>
+      {showAddFilePopUp && (
+        <div className="absolute left-0 top-0 z-50 flex h-screen w-screen flex-row place-content-center items-start justify-center backdrop-blur-[2px] backdrop-brightness-75">
```

Comment:
Styling looks a bit off on a big screen. It's not a priority right now, but worth looking into! Additionally,  we probably want to have the popup have a fixed height and scroll on a smaller height screen

![image](https://github.com/JumboCode/TheLegacyProject2/assets/74647679/d37292b0-fe62-4df1-87d0-ae87c8bf97a8)

![image](https://github.com/JumboCode/TheLegacyProject2/assets/74647679/db22c363-a1fa-4de1-92fd-1b8547517828)